src/sagemaker/jumpstart/types.py

```diff
@@ -1732,32 +1736,50 @@ def from_json(self, json_obj: Dict[str, Any]) -> None:
         )
         self.inference_config_rankings: Optional[Dict[str, JumpStartConfigRanking]] = (
             {
-                alias: JumpStartConfigRanking(ranking)
+                alias: JumpStartConfigRanking(ranking, is_hub_content=self._is_hub_content)
                 for alias, ranking in json_obj["inference_config_rankings"].items()
             }
             if json_obj.get("inference_config_rankings")
             else None
         )
-        inference_configs_dict: Optional[Dict[str, JumpStartMetadataConfig]] = (
-            {
-                alias: JumpStartMetadataConfig(
-                    alias,
-                    config,
-                    json_obj,
-                    (
-                        {
-                            component_name: self.inference_config_components.get(component_name)
-                            for component_name in config.get("component_names")
-                        }
-                        if config and config.get("component_names")
-                        else None
-                    ),
-                )
-                for alias, config in json_obj["inference_configs"].items()
-            }
-            if json_obj.get("inference_configs")
-            else None
-        )
+
+        if self._is_hub_content:
```

Comment:
non-blocking: since we are here, it would be good if we could update line 1792 for training configs, but doesn't have to be in this PR.

Reply:
thanks for the callout, I added those changed as well